processing/src/main/java/io/druid/query/dimension/CascadeExtractionDimensionSpec.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.dimension;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Preconditions;
+import com.metamx.common.StringUtils;
+import io.druid.query.extraction.CascadeExtractionFn;
+import io.druid.query.extraction.ExtractionFn;
+import io.druid.segment.DimensionSelector;
+
+import java.nio.ByteBuffer;
+
+/**
+ */
+public class CascadeExtractionDimensionSpec implements DimensionSpec
```

Comment:
Would it be cleaner/simpler to have a CascadeExtractionFn that defines the subfunctions as a JSON array, to be chained in the order in which functions are defined in the array, instead of using a child field?

Then we could avoid adding a new DimensionSpec and the user could still specify chained ExtractionFns as an array